helix-view/src/editor.rs

```diff
@@ -238,9 +238,28 @@ impl Editor {
                     self.documents[view.doc].selection(view.id).clone(),
                 );
 
+                let doc = doc!(self);
+                // If the current view is an empty scratch buffer and is not displayed in any other views, delete it.
+                // Boolean value is determined before the creation of the `view` variable because the operation
+                // requires a borrow of `self.tree`, which is mutably borrowed when `view` is created.
+                let remove_empty_scratch = !doc.is_modified()
+                    // If the buffer has no path and is not modified, it is an empty scratch buffer.
+                    && doc.path().is_none()
+                    // Ensure the buffer is not displayed in any other splits.
+                    && !self
+                        .tree
+                        .traverse()
+                        .any(|(_, view)| view.doc == doc.id && view.id != view!(self).id);
                 let view = view_mut!(self);
                 view.jumps.push(jump);
```

Comment:
You only want this to happen if `!remove_empty_scratch`, we push the old position onto the jumplist so you can `ctrl-o` back to it. If you push it here then close the document you'll run into issues.

Reply:
Ah, I didn't actually know what that did.